pennylane/ops/qubit/matrix_ops.py

```diff
@@ -193,6 +196,8 @@ def compute_matrix(U: TensorLike):  # pylint: disable=arguments-differ
         [[0.98877108+0.j, 0.-0.14943813j],
         [0.-0.14943813j, 0.98877108+0.j]]
         """
+        if sp.sparse.issparse(U):
+            return U.toarray()
```

Comment:
I'm kinda going back and forth about whether or not we should `raise MatrixUndefinedError` here if `U` is sparse. I am leaning toward only providing a dense matrix `U` is dense and only provide a sparse matrix if `U` is sparse. 

Then it's up to whatever consumes the operator to doing any dense-sparse or sparse to dense conversions.

Mind giving that a try and seeing what happens?

Reply:
Trying here: https://github.com/PennyLaneAI/pennylane/pull/6986/commits/7030e316db1d0eef4338dbb135c145e0be77bfe0

Reply:
Where the failures just unit tests?

Reply:
@albi3ro vastly most are `Controlled.sparse_matrix(wire_order=..)` not implemented.

Reply:
which is still interested though. Why before I've never seen them triggered?

Reply:
Why would changing `QubitUnitary` have changed `Controlled` tests?

Reply:
@albi3ro it seems that the deferred measurement converts some Conditionals into Controlled, and then the Controlled used Controlled(QubitUnitary(...))

Reply:
> @albi3ro it seems that the deferred measurement converts some Conditionals into Controlled, and then the Controlled used Controlled(QubitUnitary(...))

This can be observed by putting break point between `results1` and `results2` inside `tests/devices/qubit/test_simulate.py::TestMidMeasurements::test_simple_dynamic_circuit`, at commit https://github.com/PennyLaneAI/pennylane/pull/6986/commits/24a8f1b3d990cc362450450cb6bc71a67d82bf4a